app/models/user.js

```diff
@@ -109,34 +108,32 @@ var User = DS.Model.extend({
         let promises = schoolIds.map(id => {
           return store.findRecord('school', id);
         });
-        RSVP.all(promises).then(schools => {
+        all(promises).then(schools => {
           schools.pushObject(primarySchool);
-          defer.resolve(schools.uniq());
+          deferred.resolve(schools.uniq());
         });
       });
     });
     return PromiseArray.create({
```

Comment:
PromiseArray is a private API of ember-data - while cleaning this up I think switching to returning a plain Promise is probably a good idea.